go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -223,149 +268,89 @@ func (d *DoltSession) ValidateSession(ctx *sql.Context, dbName string) error {
 	if d.validateErr != nil {
 		return d.validateErr
 	}
-	sessionState, ok, err := d.LookupDbState(ctx, dbName)
-	if err != nil {
-		return err
-	}
-	if !ok {
-		return nil
-	}
-	if sessionState.WorkingSet == nil {
-		return nil
-	}
-	wsRef := sessionState.WorkingSet.Ref()
-	_, err = sessionState.dbData.Ddb.ResolveWorkingSet(ctx, wsRef)
-	if err == doltdb.ErrWorkingSetNotFound {
-		_, err = d.newWorkingSetForHead(ctx, wsRef, dbName)
-		// if the current head is not found, the branch was force deleted, so use nil working set.
-		if errors.Is(err, doltdb.ErrBranchNotFound) {
-			return ErrCurrentBranchDeleted
-		} else if err != nil {
-			return err
-		}
-	} else if err != nil {
-		return err
-	}
 	return nil
```

Comment:
(super minor) You could just return `d.validateErr` directly instead of checking for `nil`